Dockerfile

```diff
@@ -16,27 +19,56 @@ RUN /src/scripts/docker-package.sh
 # Copy the config now so that we don't create another layer in the app image
 RUN cp /src/config.sample.json /src/webapp/config.json
 
-# App
-FROM nginx:alpine-slim
+FROM nginxinc/nginx-unprivileged:${DEBIAN_VERSION} AS nginx
 
 COPY --from=builder /src/webapp /app
 
 # Override default nginx config. Templates in `/etc/nginx/templates` are passed
 # through `envsubst` by the nginx docker image entry point.
 COPY /docker/nginx-templates/* /etc/nginx/templates/
 
-# Tell nginx to put its pidfile elsewhere, so it can run as non-root
-RUN sed -i -e 's,/var/run/nginx.pid,/tmp/nginx.pid,' /etc/nginx/nginx.conf
-
-# nginx user must own the cache and etc directory to write cache and tweak the nginx config
-RUN chown -R nginx:0 /var/cache/nginx /etc/nginx
-RUN chmod -R g+w /var/cache/nginx /etc/nginx
-
+USER root
 RUN rm -rf /usr/share/nginx/html \
   && ln -s /app /usr/share/nginx/html
 
+# Run a no-op action of nginx to run entrypoint scripts that may tweak config files
+USER nginx
+RUN ["nginx", "-t"]
```

Comment:
This won't work as this isn't the final image, yet uses runtime environment variables.

https://github.com/element-hq/element-web/blob/develop/docs/install.md#docker specifies `ELEMENT_WEB_PORT` runtime envvar, which is consumed by the nginx entrypoint scripts which tweak the nginx config files, if this is run only at build time then it will be broken.